esp/esp/dbmail/cronmail.py

```diff
@@ -45,7 +45,7 @@
 
 import os
 
-@transaction.autocommit
+@transaction.atomic
```

Comment:
Interesting question.  I think what we want is `atomic` on the first one, and then the second one should either autocommit, or have `atomic` wrappers somewhere further inside (e.g. around the `mailtxt.send()`).  This way, if `MessageRequest` processing fails, we just back out entirely, because that's all happening in the DB, but if `TextOfEmail` processing fails, we keep whatever we've done already, so that we don't hit the people we've already emailed again, if we can avoid it.

Whichever ones we switch to `atomic` should also have some of their exception-catching removed, because catching exceptions can prevent the transaction from rolling back correctly.  So it may be time to just roll a fix for #1666 into this PR.